src/app/snap_to_grid.cpp

```
@@ -11,18 +11,108 @@
+  const double halfW = tileW / 2.0;
+  const double halfH = tileH / 2.0;
+
+  // Grid origin in screen coordinates
+  const double originX = grid.x;
+  const double originY = grid.y;
+
+  // Convert screen point to tile coordinates (relative to origin)
+  const double relX = point.x - originX;
+  const double relY = point.y - originY;
+  
+  // Screen to tile transformation (inverse of diamond projection)
```

Cppcheck (2.19.0)

[error] Division by zero.

(zerodiv)